api/context/go121_context_compat.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2014 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+package context
```

Comment:
I wonder if we can make this an internal package. It's not really considered part of our public API so if we can avoid exporting it outside of teleport that would be ideal.

Reply:
+1 for not exporting if we can, but I'm not sure we can nest a package under api/internal and still have it visible for lib/.

Reply:
It _might_ work since we nest the API in the teleport codebase.

Alternatively, we could do 2 internal packages (I don't love it..)

Reply:
Yeah, I considered double-copying it too. It isn't great but it's a copy anyway.

I don't think nesting under api/internal works though.

Reply:
Seems to need 2 copies unfortunately